kybra/compiler/kybra_post_install/src/main.rs

```diff
@@ -0,0 +1,402 @@
+    let (canister_id, canister_already_its_own_controller) =
+        add_permissions(canister_name, &dfx_network)?;
+    install_app_canister(canister_name, &dfx_network)?;
+    remove_permissions(
+        canister_name,
+        &dfx_network,
+        &canister_id,
+        canister_already_its_own_controller,
+    )?;
+
+    // TODO this is here because of some complications with the install_code self-referential cross-canister call
+    // TODO the call is a notify and thus won't wait for the canister's post_upgrade function to complete
+    // TODO we wait here to make sure that the canister is most likely initialized before ending the post_install script
+    thread::sleep(Duration::from_secs(5));
+
+    Ok(())
+}
+
+fn upload_app_canister(
+    canister_name: &str,
+    max_chunk_size: usize,
+    dfx_network: &str,
+) -> Result<(), String> {
+    let wasm = std::fs::read(format!(
+        ".kybra/{canister_name}/{canister_name}_app.wasm.gz"
+    ))
+    .map_err(|e| error_to_string(&e))?;
+
+    let wasm_chunks = split_into_chunks(wasm, max_chunk_size);
+
+    for (index, wasm_chunk) in wasm_chunks.iter().enumerate() {
+        upload_chunk(
+            &format!("{canister_name} wasm"),
+            canister_name,
+            wasm_chunk,
+            "upload_wasm_chunk",
+            dfx_network,
+            index,
+            wasm_chunks.len(),
+        )?;
+    }
+
+    Ok(())
+}
+
+fn upload_python_stdlib(
+    canister_name: &str,
+    max_chunk_size: usize,
+    dfx_network: &str,
+) -> Result<(), String> {
+    let python_stdlib_bytecode = get_python_stdlib_bytecode()?;
+
+    let local_python_stdlib_bytecode_hash =
+        get_local_python_stdlib_bytecode_hash(&python_stdlib_bytecode);
+
+    let remote_python_stdlib_bytecode_hash =
+        get_remote_python_stdlib_bytecode_hash(canister_name, dfx_network)?;
+
+    let hashes_do_not_match =
+        format!("(\"{local_python_stdlib_bytecode_hash}\")") != remote_python_stdlib_bytecode_hash;
+
+    if hashes_do_not_match {
+        handle_python_stdlib_hashes_do_not_match(
+            canister_name,
+            dfx_network,
+            python_stdlib_bytecode,
+            max_chunk_size,
+        )?;
+    }
+
+    Ok(())
+}
+
+fn get_local_python_stdlib_bytecode_hash(python_stdlib_bytecode: &Vec<u8>) -> String {
+    let mut hasher = Sha256::new();
+    hasher.update(python_stdlib_bytecode);
+    let result = hasher.finalize();
+    hex::encode(result)
+}
+
+fn get_remote_python_stdlib_bytecode_hash(
+    canister_name: &str,
+    dfx_network: &str,
+) -> Result<String, String> {
+    let remote_python_stdlib_hash_output = Command::new("dfx")
+        .arg("canister")
+        .arg("call")
+        .arg("--network")
+        .arg(dfx_network)
+        .arg(canister_name)
+        .arg("python_stdlib_hash")
+        .output()
+        .map_err(|_| create_error_string("Failed to execute the dfx canister call command"))?;
```

Comment:
I'd maybe specify that you were trying to call the `python_stdlib_hash` function on your canister, or say what you were doing in general for more context.